pkg/apply/apply.go

```diff
@@ -90,7 +90,7 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 	if err := MergeObjectForUpdate(existing, obj); err != nil {
 		return errors.Wrapf(err, "could not merge object %s with existing", objDesc)
 	}
-	if !equality.Semantic.DeepEqual(existing, obj) {
+	if !equality.Semantic.DeepDerivative(obj, existing) {
```

Comment:
So parameter order doesn't matter ?

Reply:
DeepDerivative will ignore the fields which are not set in the first object. `obj` only contains the fields set by the controller intentionally.